torch/_custom_op.py

```diff
@@ -370,3 +447,61 @@ def validate_function_matches_schema(
             f"Schema has kwonlyarg names {kwonlyarg_names} but function has "
             f"{arg_spec.kwonlyargs}."
         )
+
+
+# Global dictionary holding weak references to all CustomOp objects
+# Used to query the CustomOp associated with a specific C++ dispatcher operator.
+# An example usage is FakeTensor: FakeTensor checks if a specific operator
+# has an implementation registered via the CustomOp API.
+global_registry: typing.Dict["qualname", "weakref.ref[CustomOp]"] = {}
```

Comment:
weakref.WeakValueDictionary?

Reply:
Didn't know that existed, looks like that is what I am looking for. Thanks!